stdlib/test/builtin/test_string.mojo

```diff
@@ -222,35 +223,40 @@ fn test_atol() raises:
     assert_equal(-52, atol(String(" -52")))
     assert_equal(-69, atol(String(" -69  ")))
 
+    # other bases
+    assert_equal(10, atol("A", 16))
+    assert_equal(15, atol("f ", 16))
+    assert_equal(255, atol(" FF", 16))
+    assert_equal(18, atol("10010", 2))
+    assert_equal(35, atol("Z", 36))
+
     # Negative cases
-    try:
+    with assert_raises(
+        contains="String is not convertible to integer with base 10."
+    ):
         _ = atol(String("9.03"))
-        raise Error("Failed to raise when converting string to integer.")
-    except e:
-        assert_equal(str(e), "String is not convertible to integer.")
 
-    try:
+    with assert_raises(contains="String is not convertible to integer."):
         _ = atol(String(" 10 1"))
-        raise Error("Failed to raise when converting string to integer.")
-    except e:
-        assert_equal(str(e), "String is not convertible to integer.")
 
-    try:
+    with assert_raises(
+        contains="String is not convertible to integer with base 5."
```

Comment:
**Suggestion** I think it would be good to include the original user's requested string in this error message — what do you think?  For example, the error from Python is:

``` python
int("5", 5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 5: '5'
```

If we do this, I think we shouldn't special case the empty string error message either.  It's consistent in Python, for example:

``` python
int("")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: ''
```

Reply:
Sure seems like a good idea